Annotations/annotation_generator.py

```diff
@@ -0,0 +1,392 @@
+    struct_name = record_decl_chunks[len(record_decl_chunks) - 2]
+
+    return specifier_manger.add_struct(struct_name)
+
+def add_parameter(parm_var_decl: str, param_index: int, func: Specifier):
+    if type(func) != Function:
+        return
+    
+    logout(f"adding param {parm_var_decl}")
+
+    param_type = parm_var_decl[
+        parm_var_decl.find("'")  + 1 : find_second_index(parm_var_decl, "'") 
+    ]
+
+    param = Parameter(param_index, param_type)
+    func.add_parameter(param)
+
+def add_field(field_decl: str, field_index: int, struct: Specifier):
+    if type(struct) != Struct: 
+        return
+
+    start_index = field_decl.find("'") - 2
+    cur_char = field_decl[start_index]
+    field_name = ""
+
+    while (cur_char != " "):
+        field_name = cur_char + field_name
+
+        start_index -= 1
+        cur_char = field_decl[start_index]
+    
+
+    field = Field(field_name, field_index)
+    struct.add_field(field)
+
+
+# finds the index of the 2nd occurrence of char in string 
+def find_second_index(string: str, char: str) -> int:
+    return string.find(char, string.find(char) + 1)
+
+# param_index, field_index are optional; can be either None or int 
+def parse_anno(anno: str, spec: Specifier, annotation_manager: AnnotationManager, param_index: int = None, field_index: int = None):
+    logout(f"pre is {anno}")
+
+    start_anno_index = anno.find('"')
+    end_anno_index = anno.rfind('"')
+
+    anno = anno[start_anno_index + 1 : end_anno_index]
+
+    anno_type = anno[
+        anno.find(" ") + 1 : find_second_index(anno, " ")
+    ]
+
+    anno_methods = anno[anno.rfind("methods = ") + len("methods = ")::]  
+    logout(f"ANNO METHODS = {anno_methods}")
+
+    known_target = ""
+
+    anno_unfilled_target = ""
+    anno_unfilled_target_index = find_second_index(anno, "_")
+
+    while (anno[anno_unfilled_target_index] != " "):
+        anno_unfilled_target += anno[anno_unfilled_target_index]
+        anno_unfilled_target_index += 1
+    
+    # anno_unfilled_target looks like "_" or "_.FIELD(x)"
+
+    if type(spec) is Struct:
+        known_target = f"STRUCT({spec.get_name()}).FIELD({field_index})"
+        
+
+    elif type(spec) is Function:
+        known_target = f"FUNCTION({spec.get_name()})"
+
+        # Assumption: if it is not referring to a parameter, it is 
+        # referring to the function itself 
+        if param_index is None: 
+            known_target += ".RETURN"
+        else: 
+            known_target += f".PARAM({param_index})"
+
+
+    if ".FIELD" in anno_unfilled_target:
+            anno_unfilled_target_field_name = anno_unfilled_target[
+                anno_unfilled_target.find("(") + 1 : anno_unfilled_target.find(")")
+            ]
+            
+            if type(spec) is Function:
+                if param_index is None:
+                    logout(f"ret type is '{spec.return_type}'")
+                    # Assumption: spec.return_type looks like 'struct <struct_name> '
```

Comment:
Why can't `spec.return_type` be a primitive (e.g., `int`)?

Reply:
Adding the following comment: 
```
               # Assumption: spec.return_type looks like 'struct <struct_name> '
                    # we can make this assumption because: 
                    # 1. param_index is none, which means we are making an 
                    # annotation for the function itself, and (for now) we assume
                    # that means it's an annotation on its return.
                    # note: this will need to update
                    # if/when annotations like @Owning and @MustCallAlias are implemented
                    # 2. the ".FIELD" in anno_unfilled_target is specified by the user via 
                    # something like 
                    '''
                    char *MustCall("free") 
                        creates_obligation(char *s Calls("free"), struct my_struct X Calls("free", "x"));
                                                                                                   ^^^
                    '''
                    # the calling of that second parameter on Calls/MustCall tells us that for the struct 
                    # we are specifying, we are targeting some field. we verify whether that field 
                    # actually exists, since this pass does
                    # keep note of every struct and its fields. if we wrote something like:
                    # void Calls("free", "x") example(int S Calls("free", "x")) { ... }
                    #                                 ^^^                 ^^^ implying "int" has field "x"
                    # we'll throw "ValueError: Did not find field 'x' for struct 'int', anno TOOL_CHECKER Calls target = _.FIELD(x) methods = free"
```

Reply:
This looks good, with one note:

> # note: this will need to update
> # if/when annotations like @Owning and @MustCallAlias are implemented

Please describe what update will be needed.